collect_app/src/main/java/org/odk/collect/android/utilities/WebUtils.java

```diff
@@ -248,22 +248,12 @@ public static synchronized HttpClient createHttpClient(int timeout) {
     public static void discardEntityBytes(HttpResponse response) {
         HttpEntity entity = response.getEntity();
         if (entity != null) {
-            InputStream is = null;
-            try {
-                is = response.getEntity().getContent();
-                while (is.read() != -1) {
+            try (InputStream is = entity.getContent()) {
+                while (is.available() != 0 && is.read() != -1) {
```

Comment:
The returned value from is.available() can also be used to read larger chunks, rather than looping through every byte

Reply:
Does this seem better?

```
                byte[] buffer = new byte[1024];
                while (is.available() != 0) {
                    is.read(buffer, 0, is.available());
                }

```